ts/screens/profile/NewEmailInsertScreen.tsx

```diff
@@ -0,0 +1,294 @@
+            EMPTY_EMAIL === value ||
+            !validator.isEmail(value) ||
+            areSameEmails
+          ) {
+            return undefined;
+          }
+          return E.isRight(EmailString.decode(value));
+        }),
+        O.toUndefined
+      ),
+    [areSameEmails, email]
+  );
+
+  const continueOnPress = () => {
+    Keyboard.dismiss();
+
+    pipe(
+      email,
+      O.map(e => {
+        updateEmail(e as EmailString);
+      })
+    );
+  };
+
+  const renderFooterButtons = () => {
+    const continueButtonProps = {
+      disabled: !isValidEmail() && !isLoading,
+      onPress: continueOnPress,
+      title: I18n.t("global.buttons.continue"),
+      block: true,
+      primary: isValidEmail()
+    };
+
+    return (
+      <FooterWithButtons
+        type={"SingleButton"}
+        leftButton={continueButtonProps}
+      />
+    );
+  };
+
+  const handleOnChangeEmailText = (value: string) => {
+    setAreSameEmails(areStringsEqual(O.some(value), optionEmail, true));
+    setEmail(value !== EMPTY_EMAIL ? O.some(value) : O.none);
+  };
+
+  const handleGoBack = useCallback(() => {
+    // goback if the onboarding is completed
+    props.navigation.goBack();
+  }, [props.navigation]);
+
+  useEffect(() => {
+    setAreSameEmails(false);
+    setEmail(O.some(EMPTY_EMAIL));
+  }, []);
+
+  const prevUserProfile = usePrevious(profile);
+
+  useEffect(() => {
+    if (prevUserProfile) {
+      const isPrevCurrentSameState = prevUserProfile.kind === profile.kind;
+      // do nothing if prev profile is in the same state of the current
+      if (isPrevCurrentSameState) {
+        return;
+      }
+    }
+  }, [prevUserProfile, profile]);
+
+  useEffect(() => {
+    if (prevUserProfile && pot.isUpdating(prevUserProfile)) {
+      if (pot.isError(profile)) {
+        // the user is trying to enter an email already in use
+        if (profile.error.type === "PROFILE_EMAIL_IS_NOT_UNIQUE_ERROR") {
+          Alert.alert(
+            I18n.t("email.insert.alertTitle"),
+            I18n.t("email.insert.alertDescription"),
+            [
+              {
+                text: I18n.t("email.insert.alertButton"),
+                style: "cancel"
+              }
+            ]
+          );
+        } else {
+          showToast(I18n.t("email.edit.upsert_ko"), "danger");
+        }
+
+        // display a toast with error
+      } else if (pot.isSome(profile) && !pot.isUpdating(profile)) {
+        // the email is correctly inserted
+        if (isEmailValidated) {
+          handleGoBack();
+        } else {
+          showModal(<NewRemindEmailValidationOverlay />);
+        }
+        return;
+      }
+    }
+  }, [handleGoBack, isEmailValidated, prevUserProfile, profile, showModal]);
+
+  return (
+    <LoadingSpinnerOverlay isLoading={isLoading}>
+      <BaseScreenComponent
+        goBack={handleGoBack}
+        headerTitle={I18n.t("profile.data.list.email")}
+        contextualHelpMarkdown={contextualHelpMarkdown}
+      >
```

Comment:
What about extracting the same UI in a separate component and using it in both `NewEmailIinsertScreen` and `NewOnboardingEmailInsertScreen`? This way, if this UI changes we add changes only in a single point. What do you think about that?

Reply:
Since the task is blocking I added a "fixme" so we can try it later. I've tried it and I don't think it's a good idea because the logic becomes quite convoluted and full of ternaries controls on the ux/render part. Anyway, my idea, to be discussed in an eventual refactor, would be to try to put the logic together and not the render.